src/utils/Firebase/firebase.js

```diff
@@ -214,46 +227,44 @@ class Firebase {
             console.error("Error creating custom lesson: ", error)
           );
 
-        if (lessonDocId) {
-          // Push custom lessons to deployments
-          let batch = this.db.batch();
-          let deploymentAccountId;
+        // Push custom lesson to deployment accounts
+        let batch = this.db.batch();
 
-          // Add/update deploymentAccount docs
-          for (deploymentAccountId of deploymentAccountIds) {
+        // Add/update deploymentAccount references to this custom lesson
+        for (const deploymentAccountId of deploymentAccountIds) {
+          let deploymentRef = this.db.doc(
+            `deployment_account/${deploymentAccountId}/`
+          );
+          batch.update(deploymentRef, {
+            customLessons: app.firestore.FieldValue.arrayUnion(
+              customLessonRef.id
+            )
+          });
+        }
+
+        // Remove custom lesson for non-assigned students
+        for (let deploymentAccountId of currentAssignedDeploymentIds) {
+          if (!deploymentAccountIds.includes(deploymentAccountId)) {
             let deploymentRef = this.db.doc(
               `deployment_account/${deploymentAccountId}/`
             );
             batch.update(deploymentRef, {
-              customLessons: app.firestore.FieldValue.arrayUnion(
+              customLessons: app.firestore.FieldValue.arrayRemove(
                 customLessonRef.id
               )
             });
           }
-
-          for (let deploymentAccountId of currentAssignedDeploymentIds) {
-            if (!deploymentAccountIds.includes(deploymentAccountId)) {
-              let deploymentRef = this.db.doc(
-                `deployment_account/${deploymentAccountId}/`
-              );
-              batch.update(deploymentRef, {
-                customLessons: app.firestore.FieldValue.arrayRemove(
-                  customLessonRef.id
-                )
-              });
-            }
-          }
-
-          return batch
-            .commit()
-            .catch(error =>
-              console.error(
-                "Pushing custom lessons to deployments failed: ",
-                error
-              )
-            );
-          // needs better promise rejection
         }
+
+        return batch
+          .commit()
+          .catch(error =>
+            console.error(
+              "Pushing custom lessons to deployments failed: ",
+              error
+            )
+          );
+        // needs better promise rejection
```

Comment:
elaborate on what this means please

Reply:
hello!!

Reply:
oh shit this is outdated lmao this doesn't matter anymore my b

Reply:
ill remove this